### PR TITLE
server: reduce allocations in setupSpanForIncomingRPC

### PR DESCRIPTION
**Pull request by andreimatei**
A couple of variables were independently escaping because they were
captured by a closure. This patch groups them into a single allocation,
saving two allocations per call.

name                        old time/op    new time/op    delta
SetupSpanForIncomingRPC-32     508ns ± 2%     442ns ± 0%  -12.91%  (p=0.008 n=5+5)
name                        old alloc/op   new alloc/op   delta
SetupSpanForIncomingRPC-32      178B ± 0%      145B ± 0%  -18.54%  (p=0.008 n=5+5)
name                        old allocs/op  new allocs/op  delta
SetupSpanForIncomingRPC-32      4.00 ± 0%      2.00 ± 0%  -50.00%  (p=0.008 n=5+5)

Release note: None

### CODE REVIEW

**Review by abarganier — APPROVED** [review]
:lgtm: once merge conflict is fixed.

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/76174#-:-MvUdVGI9Aj4iiP9sVvl:boxxfay)* status: :shipit: complete! 1 of 0 LGTMs obtained

**Review by andreimatei — COMMENTED** [review]
TFTR!

bors r+

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/76174#-:-MvUfYzL6du6ulbc8ZPb:b-oduykh)* status: :shipit: complete! 0 of 0 LGTMs obtained (and 1 stale)

**Comment by andreimatei** [comment]
bors r-

merge skew

**Review by andreimatei — COMMENTED** [review]
bors r+

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/76174#-:-MvVD-AoAGuY8m_F0MhN:b-ydi9ay)* status: :shipit: complete! 0 of 0 LGTMs obtained (and 1 stale)

**Pull request merged**
